libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,276 @@
+          infectionRate: selectedCounty[0]['infection_rate'],
+          deaths: selectedCounty[0]['deaths'],
+          deathRate: selectedCounty[0]['death_rate']
+        });
+      });
+
+      map.on('mousemove', 'covid-state', (e) => {
+        const feature = e.features[0];
+        const selectedState = this.stateData.filter((state) => state.STATE === feature.properties.STATE);
+
+        this.infoBoxModel.next({
+          name: feature.properties.NAME,
+          pop: feature.properties.POPESTIMATE2019,
+          cases: selectedState[0]['confirmed'],
+          infectionRate: selectedState[0]['infection_rate'],
+          deaths: selectedState[0]['deaths'],
+          deathRate: selectedState[0]['death_rate']
+        });
+      });
+    });
+  }
+
+  public reloadData() {
+    const currentDateSelected = '2020-06-06';
+    console.log(currentDateSelected);
+    const stateURL =
+      'https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/states/' +
+      currentDateSelected +
+      '.json';
+    const countyURL =
+      'https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/counties/' +
+      currentDateSelected +
+      '.json';
+
+    const requests = forkJoin([this.http.get(stateURL), this.http.get(countyURL)]);
+    requests.pipe(take(1)).subscribe(([sData, cData]) => {
+      this.stateData = sData[currentDateSelected];
+      this.countyData = cData[currentDateSelected];
+      this.selectedDate = currentDateSelected;
+      this.mortalButtonToggled ? this.drawDeathMap() : this.drawCasesMap();
+    });
+  }
+
+  public drawCasesMap() {
+    const stateExpression = ['match', ['get', 'STATE']];
+    const countyExpression = ['match', ['get', 'fips']];
+
+    this.stateData.forEach(function(row) {
+      const number = row['infection_rate'];
+      const color =
+        number > 1000
+          ? '#AE8080'
+          : number > 500
+          ? '#D18080'
+          : number > 100
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      stateExpression.push(row['STATE'], color);
+    });
+
+    this.countyData.forEach(function(row) {
+      const number = row['infection_rate'];
+      const color =
+        number > 1000
+          ? '#AE8080'
+          : number > 500
+          ? '#D18080'
+          : number > 100
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      countyExpression.push(row['fips'], color);
+    });
+
+    stateExpression.push('rgba(255,255,255,1)');
+    countyExpression.push('rgba(255,255,255,1)');
+
+    this.mapService.loaded.pipe(take(1)).subscribe((map) => {
+      map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+      map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+    });
```

Comment:
You can also use the `map` instance directly from the `mapService` much like you did in the `switchLayers` toggle.

Reply:
resolved